src/certificate.js

```diff
@@ -55,14 +66,29 @@ function Certificate(certificate) {
   this.certificate = certificate;
```

Comment:
The input `certificate` needs to be pretty well-formed or some of your other functions might get `undefined` errors when trying to refer to its keys. TypeScript might help here.

Also, we should probably build some kind of "builder" to allow someone to programmatically build a certificate from scratch.

Reply:
Also, I suggest the following changes to this class:

- Wrap the passed in certificate as [immutable](https://facebook.github.io/immutable-js/docs/#/)
- Every time you need to mutate the object (e.g. privacy filter, deep clone the certificate, mutate it, then return a new `Certificate` object with the modified certificate.